### PR TITLE
improves coverage and API of httpRequest

### PR DESCRIPTION
**Pull request by flovilmart**
Bumps code coverage > 90% :tada:

### CODE REVIEW

File: src/httpRequest.js [hunk 1]
```diff
@@ -9,13 +9,15 @@ module.exports = function(options) {
   };
   delete options.success;
   delete options.error;
-  if (options.uri && !options.url) {
-    options.uri = options.url;
-    delete options.url;
-  }
+  delete options.uri; // not supported
```
Comment (drew-gross) [inline src/httpRequest.js]: You sure this isn't going to break anything?

File: src/httpRequest.js [hunk 2]
```diff
@@ -9,13 +9,15 @@ module.exports = function(options) {
...
   if (typeof options.body === 'object') {
     options.body = JSON.stringify(options.body);
+    options.headers = options.headers || {};
+    options.headers['Content-Type'] = "application/json";
   }
+  // set follow redirects to false by default
+  options.followRedirect = options.followRedirects == true ? true : false;
```
Comment (drew-gross) [inline src/httpRequest.js]: This seems to have snuck in by accident I think

**Comment by drew-gross** [comment]
`request` has a `json` option that might help simplify some of these tests, fyi.

File: src/httpRequest.js (hunk 2 above)
Comment (flovilmart) [inline src/httpRequest.js]: This is to match Parse.Cloud.httpRequest documentation vs request.

File: src/httpRequest.js (hunk 1 above)
Comment (flovilmart) [inline src/httpRequest.js]: Uri is not supported in Parse.Cloud.httpRequest AFAIK

**Comment by flovilmart** [comment]
For the body: ```The body of the request. If it is a JSON object, then the Content-Type set in the headers must be application/x-www-form-urlencoded or application/json. You can also set this to a {@link Buffer} object to send raw bytes. If you use a Buffer, you should also set the Content-Type header explicitly to describe what these bytes represent.``` 
I should not set the content type, and provide an URL encoding of the body if the header is `application/x-www-form-urlencoded` 
This was not supported before.

File: src/httpRequest.js (hunk 2 above)
Comment (flovilmart) [inline src/httpRequest.js]: http://www.parse.com/docs/js/api/classes/Parse.Cloud.HTTPOptions.html

Comment (drew-gross) [inline src/httpRequest.js]: Sorry, I was referring to the `? true : false;` part, which is not necessary.

File: src/httpRequest.js (hunk 1 above)
Comment (drew-gross) [inline src/httpRequest.js]: :+1:

File: src/httpRequest.js (hunk 2 above)
Comment (flovilmart) [inline src/httpRequest.js]: You're right :)

**Pull request merged**